Add CI/CD and IaC to the glossary (#1563)
hashicorp/terraform-website: content/source/docs/glossary.html.md

A repository usually has a default branch (whose name, in [Git][], defaults to `master`), which successful changes are eventually merged into. Most modern development workflows also include topic branches (where a specific set of changes is explored, iterated on, and verified), and some workflows include long-lived branches other than the default branch.

## CI/CD

[ci-cd]: glossary.html#ci-cd

> CI/CD (Continuous Integration/Continuous Delivery) platforms provide automated, structured pipelines for new releases. Infrastructure management, typically handled by operations teams, can also fit into an automated release workflow using infrastructure as code with Terraform.

Reviewer: Hmm, this says something about the "why" of CI/CD, but kind of evades the "what." (When I was new in tech, this was the type of explanation that contributed the most to my impostor syndrome.) I'd like to find a way to cram an explanation in here somewhere. 

I usually see people using CI/CD to mean one or both of the following:

- A computer program, running ✨somewhere✨, that performs pre-defined tasks whenever it notices that certain events have occurred. Usually this translates to automatically running tests and building deployable artifacts whenever new commits get pushed. 
- The practice (independent of the actual tooling) of trying to do all your releasing via automated, structured pipelines rather than manual interactive processes. 

IDK, any thoughts about how to get some of that in there? My brain might have exited the "make words good" portion of the day. But at any rate, I think it'd be kind to newbies trying to get a handle on this space if we could get the part about "program that reacts to events" somewhere in there.

Author: Maybe this? 

```suggestion
CI/CD (Continuous Integration/Continuous Delivery) platforms provide automated, structured pipelines for new releases. They usually do this by performing pre-defined tasks such as tests or builds in response to trigger events like commits. Infrastructure management, typically handled by operations teams, can also fit into an automated release workflow using infrastructure as code with Terraform.
```

Reviewer: Hmm, how do you feel about swapping the order? 

```suggestion
CI/CD (Continuous Integration/Continuous Delivery) platforms perform pre-defined tasks (like tests or builds) in response to trigger events (like new commits in a project). This helps make new releases more automated, predictable, and understandable. Infrastructure management, typically handled by operations teams, can also fit into an automated release workflow using infrastructure as code with Terraform.
```

You can usually copy an external ID from the URL bar when viewing an object in Terraform Cloud's UI. Workspaces don't display an ID in the URL bar, but their general settings page includes a UI control for viewing and copying the ID.

## Infrastructure as Code

[infrastructure-as-code]: glossary.html#infrastructure-as-code

> Infrastructure-as-Code (IaC) is the process of managing infrastructure in a file or files rather than manually configuring it via a user interface. Infrastructure resource types managed with IaC can include virtual machines, security groups, network interfaces, and many others.

Reviewer: This feels more like an ongoing circular "practice" than a beginning/middle/end "process"... y/n?

```suggestion
Infrastructure-as-Code (IaC) is the practice of managing infrastructure in a file or files rather than manually configuring it via a user interface. Infrastructure resource types managed with IaC can include virtual machines, security groups, network interfaces, and many others.
```
